astropy/modeling/core.py

```diff
@@ -1061,8 +1061,9 @@ def _validate_input_shapes(self, inputs, argnames, model_set_axis):
                 )
             )
 
-        input_shape = check_broadcast(*all_shapes)
-        if input_shape is None:
+        try:
+            input_shape = check_broadcast(*all_shapes)
+        except IncompatibleShapeError:
             raise ValueError(
                 "All inputs must have identical shapes or must be scalars."
             )
```

Comment:
```suggestion
        except IncompatibleShapeError as err:
            raise ValueError(
                "All inputs must have identical shapes or must be scalars."
            ) from err
```
Raising the error from the original would be good for stack tracing